packages/js/src/session/session.ts

```diff
@@ -18,24 +20,64 @@ export class Session {
   }
 
   public get applicationIdentifier() {
-    return this.#options.applicationIdentifier;
+    if ('applicationIdentifier' in this.#options) {
+      return this.#options.applicationIdentifier;
+    }
+
+    return undefined;
   }
 
   public get subscriberId() {
-    return this.#options.subscriber.subscriberId;
+    if ('subscriber' in this.#options && this.#options.subscriber) {
+      return this.#options.subscriber.subscriberId;
+    }
+
+    return undefined;
+  }
+
+  private getStoredApplicationIdentifier(): string | null {
+    if (typeof window !== 'undefined' && window.localStorage) {
+      return window.localStorage.getItem(NOVU_APP_ID_KEY);
+    }
+
+    return null;
+  }
+
+  private storeApplicationIdentifier(identifier: string): void {
+    if (typeof window !== 'undefined' && window.localStorage) {
+      window.localStorage.setItem(NOVU_APP_ID_KEY, identifier);
+    }
   }
 
   public async initialize(): Promise<void> {
     try {
-      const { applicationIdentifier, subscriberHash, subscriber } = this.#options;
+      const subscriber = 'subscriber' in this.#options ? this.#options.subscriber : undefined;
+      const subscriberHash = 'subscriberHash' in this.#options ? this.#options.subscriberHash : undefined;
+      const applicationIdentifier =
+        'applicationIdentifier' in this.#options ? this.#options.applicationIdentifier : undefined;
+
+      let finalApplicationIdentifier = applicationIdentifier;
+      if (!finalApplicationIdentifier) {
+        const storedAppId = this.getStoredApplicationIdentifier();
+        if (storedAppId) {
+          finalApplicationIdentifier = storedAppId;
+        }
+      } else {
+        this.storeApplicationIdentifier('');
```

Comment:
I wonder if it's better to delete it altogether, not crticial tho

Reply:
done